compute/main.go

```diff
@@ -0,0 +1,596 @@
+}
+
+func connectionAzure() (*armcore.Connection, error) {
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := armcore.NewDefaultConnection(cred, &armcore.ConnectionOptions{
+		Logging: azcore.LogOptions{
+			IncludeBody: true,
+		},
+	})
+	return conn, nil
+}
+
+func createResourceGroup(ctx context.Context, connection *armcore.Connection) (*armresources.ResourceGroup, error) {
+	resourceGroup := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	parameters := armresources.ResourceGroup{
+		Location: to.StringPtr(location),
+		Tags:     map[string]*string{"sample-rs-tag": to.StringPtr("sample-tag")}, // resource group update tags
+	}
+
+	resp, err := resourceGroup.CreateOrUpdate(ctx, resourceGroupName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.ResourceGroup, nil
+}
+
+func deleteResourceGroup(ctx context.Context, connection *armcore.Connection) (*http.Response, error) {
+	resourceGroup := armresources.NewResourceGroupsClient(connection, subscriptionId)
+
+	pollerResponse, err := resourceGroup.BeginDelete(ctx, resourceGroupName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func createVirtualNetwork(ctx context.Context, connection *armcore.Connection) (*armnetwork.VirtualNetwork, error) {
+	vnetClient := armnetwork.NewVirtualNetworksClient(connection, subscriptionId)
+
+	parameters := armnetwork.VirtualNetwork{
+		Resource: armnetwork.Resource{
+			Location: to.StringPtr(location),
+		},
+		Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+			AddressSpace: &armnetwork.AddressSpace{
+				AddressPrefixes: []*string{
+					to.StringPtr("10.1.0.0/16"), // example 10.1.0.0/16
+				},
+			},
+			//Subnets: []*armnetwork.Subnet{
+			//	{
+			//		Name: to.StringPtr(subnetName+"3"),
+			//		Properties: &armnetwork.SubnetPropertiesFormat{
+			//			AddressPrefix: to.StringPtr("10.1.0.0/24"),
+			//		},
+			//	},
+			//},
+		},
+	}
+
+	pollerResponse, err := vnetClient.BeginCreateOrUpdate(ctx, resourceGroupName, vnetName, parameters, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.VirtualNetwork, nil
+}
+
+func deleteVirtualNetWork(ctx context.Context, connection *armcore.Connection) (*http.Response, error) {
+	vnetClient := armnetwork.NewVirtualNetworksClient(connection, subscriptionId)
+
+	pollerResponse, err := vnetClient.BeginDelete(ctx, resourceGroupName, vnetName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := pollerResponse.PollUntilDone(ctx, 10*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func createSubnets(ctx context.Context, connection *armcore.Connection, virtualNetworkName string, subnetName string) (*armnetwork.Subnet, error) {
```

Comment:
we need to unify the signature of these methods, either they all use the global variables for `resourceGroupName`, `virtualNetworkName`, `subnetName`, or they all use parameter passed in for those values.
Same for all other functions.